### PR TITLE
Remove configuration autogeneration in Collectd.

### PR DESCRIPTION
**Pull request by dhrupadb**

### CODE REVIEW

File: collectd.conf [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Interval 60
 
+Hostname ""
```
Comment (bmoyles0117) [inline collectd.conf]: Adding the comment Igor mentioned about why this is left blank can be helpful to anyone contributing in the future, not necessarily our users.

File: deb/debian/stackdriver-agent.init [hunk 2]
```diff
@@ -179,7 +137,12 @@ d_start() {
         export https_proxy=$PROXY_URL
     fi
 
-    if ! gen_config || ! check_config; then
+    if ! check_credentials || ! check_config; then
+        log_failure_msg "not starting, credentials error"
+        return 3
+    fi
+
+    if ! check_config; then
```
Comment (bmoyles0117) [inline deb/debian/stackdriver-agent.init]: I'm not sure how this condition wouldn't be captured by the condition above.

File: collectd.conf (hunk 1 above)
Comment (dhrupadb) [reply collectd.conf]: Done.

File: deb/debian/stackdriver-agent.init (hunk 2 above)
Comment (dhrupadb) [reply deb/debian/stackdriver-agent.init]: Nope that was a bug. Fixed.

File: deb/debian/stackdriver-agent.init [hunk 3]
```diff
@@ -179,7 +137,12 @@ d_start() {
         export https_proxy=$PROXY_URL
     fi
 
-    if ! gen_config || ! check_config; then
+    if ! check_credentials; then
+        log_failure_msg "not starting, credentials error"
```
Comment (igorpeshansky) [inline deb/debian/stackdriver-agent.init]: Let's actually fold the credentials check into the config check. More specifically, it's possible to override the credentials file in the configuration, so while today we're checking for the default one, we won't always do that, and then we'd have to refactor this code -- might as well do it now.

Comment (dhrupadb) [reply deb/debian/stackdriver-agent.init]: Done.

File: rpm/stackdriver-agent [hunk 4]
```diff
@@ -61,71 +60,38 @@ if [ -n "$JAVA_LIB_DIR" ]; then
...
     # Check if the application default credentials file is in the system
     # location.
     if [ ! -f /etc/google/auth/application_default_credentials.json ]; then
         # See if the instance has the correct monitoring scopes.
         INSTANCE_SCOPES=$(curl --silent --connect-timeout 1 -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/scopes 2>/dev/null || /bin/true)
         if [ `echo "$INSTANCE_SCOPES" | grep -cE '(monitoring.write|monitoring|cloud-platform)$'` -lt 1 ]; then
-            echo >&2 "The instance has neither the application default" \
-              "credentials file nor the correct monitoring scopes; Exiting."
-            return 1
+            echo "The instance has neither the application default" \
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent]: Why did you move the redirection?

File: deb/debian/stackdriver-agent.init [hunk 5]
```diff
@@ -74,48 +73,24 @@ if [ -n "$JAVA_LIB_DIR" ]; then
...
 # return:
-#   0 if config was generated successfully.
-#   1 if there is a authentication or permissions error.
-gen_config() {
-    if [ "$AUTOGENERATE_COLLECTD_CONFIG" = "no" ]; then
-        return 0
+#   0 if config is fine
+#   1 if there is a syntax error
+#   2 if there is no configuration
+#   3 if the instance doesn't have the necessary permissions
```
Comment (igorpeshansky) [inline deb/debian/stackdriver-agent.init]: s/permissions/credentials/.

File: rpm/stackdriver-agent [hunk 6]
```diff
@@ -143,8 +109,8 @@ start () {
         export https_proxy=$PROXY_URL
     fi
 
-    if ! gen_config; then
-        echo "Not starting. Configuration error!" >&2
+    if ! check_config; then
+        echo "Not starting, configuration/credentials error." >&2
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent]: Let's prefer `echo >&2 "..."` here...

File: rpm/stackdriver-agent (hunk 4 above)
Comment (dhrupadb) [reply rpm/stackdriver-agent]: Sorry, Copy pasted from deb/stackdriver-agent.init. Fixed. 

Is there a difference between the way echo works on debian vs RHEL that affects the positioning?

File: rpm/stackdriver-agent (hunk 6 above)
Comment (dhrupadb) [reply rpm/stackdriver-agent]: Done.

File: rpm/stackdriver-agent (hunk 4 above)
Comment (igorpeshansky) [reply rpm/stackdriver-agent]: No difference -- having the redirection in the front is a readability thing (makes it obvious where the output is going without having to scroll to the end).

File: deb/debian/stackdriver-agent.init (hunk 5 above)
Comment (igorpeshansky) [reply deb/debian/stackdriver-agent.init]: Also in `rpm/stackdriver-agent`.

Comment (dhrupadb) [reply deb/debian/stackdriver-agent.init]: Done.

**Comment by igorpeshansky** [comment]
dhrupadb was at Google at the time.
